### ISSUE TITLE
suggestion: keep the api same as possible for .net core

### ISSUE THREAD

**Issue by Jack85** [aspnet/SignalR#956]
I dont know how many times you changed this, hub options, javascript proxies, 
you should keep them to be upwards compatible. You can drop things like hub state but keep the api compatible please!
Javascript proxies could be implemented, same as specifiyng endpoit for signalR not for every hub, this is hell and unnecessary !
last good version for me was 0.2.0-preview2-22683 , and if this new guy can less there is no reason to upgrade.. 
where I dind have to care about those things.. make our life  easier guys..

**Comment by NtFreX** [comment]
I have a completly different opinion. This is the price you pay when you work with an prereleased library. I do not want backward compatiblity to concepts which wheren't thought through.

The first stable release is planed for netstandard2.1 and from there on I agree with compatiblity if nothing else important comes up.

**Comment by Jack85** [comment]
This is the reason why its not usable for bigger guys, because on friday its like this and on saturday its again something else. Hope to see that feature rich and stable version from 2.1. 

If you change the api every release - like you re doing now, who is going to maintain that in bigger corporate projects? 
And you re taking features out of it... 

Similar story - MS team and those json projects (clean and readable) in .net core 1, everyone adapts on in and then guys coming with xml and csproj again. At the end big confusion in community what is really happening. 

Like we in ideal world I would like to take project in .net 4.6 change framework to .net core 2 and thats it, not being forced to change working stuff with every release of (one particular) your library. Humble opinion but from the field.

**Comment by davidfowl** [comment]
> last good version for me was 0.2.0-preview2-22683 , and if this new guy can less there is no reason to upgrade..

That version was never released to nuget.org and was never going to be supported. It was a direct port from the existing SignalR which we didn't want.

> Like we in ideal world I would like to take project in .net 4.6 change framework to .net core 2 and thats it, not being forced to change working stuff with every release of (one particular) your library. Humble opinion but from the field.

Thanks for the feedback. The API changes are *extremely* minimal on both the client and the server. The core of the Hub API on client and server are very similar. The reason things differ at all is mostly because we want to enable new scenarios or removed features entirely.

- We removed multiple hubs per end point so there's no more hub proxy
- We made the API for scaleout more flexible so that it's possible to implement different patterns that weren't possible before.
- We changed the wire protocol:
  -  to enable multiple serialization formats
  - to avoid the number of http requests to make an initial connection work
- We moved away from dynamic on the server because it isn't type safe and is slower than using Invoke with strings. Being dynamic also hides the async nature of method invocations so the compiler doesn't warn about *not* awaiting invocations (which may fail or be cancelled).
